docs/cow-protocol/tutorials/cow-swap/flash-loans.mdx

```diff
@@ -0,0 +1,264 @@
+
+# Flash loans
+
+We’ll walk through a concrete example of a debt repayment using flash loans on Sepolia.
+
+There is an [example order](https://explorer.cow.fi/sepolia/orders/0x86ff28a37bf70c549edfe753d899d44307b1b4c3ae943c6f5bdfe271942fd4f135ed9a9d1122a1544e031cc92fcc7ea599e28d9c67dc66ab) for this particular case on Sepolia, including the [corresponding on-chain transaction](https://sepolia.etherscan.io/tx/0x6f4fc9b450e08dab0ab2d935e7c7dcade9e44e330e484657e5b1fadb6c6e8851).
+The example was executed using an [automated script](https://github.com/mrnaveira/cow-flashloan-tester) that places and signs a flash loan order on the Sepolia network. The following sections of the tutorial include code snippets from this tool.
+
+## Setup
+The setup for the flash loan example is as follows:
+* A Safe wallet at address 0x35eD9A9D1122A1544e031Cc92fCC7eA599e28D9C. We use a smart contract wallet so that smart contracts can initiate the withdrawal of the collateral token.
+* The Safe borrows 10,000 USDT on Aave using 500,000 USDC as collateral.
+* Aave is used as the flash loan lender.
+* The necessary `ERC-20` approvals are set in the wallet — to CoW's Vault Relayer contract and to the relevant Aave pool contracts.
+* Some non-zero amount of the order's buy token in the Safe's balance, to pass CoW Swap order validation.
+
+The order itself intends to:
+1. Request a flash loan to Aave of 5,000 USDT.
+2. Use those 5,000 to partially repay the 10,000 USDT debt on Aave.
+3. Withdraw 400,000 USDC from Aave.
+4. Perform a trade on CoW Swap, selling up to 400,000 USDC for 5,002.5 USDT (borrowed amount + [0.05% flash loan fee on Aave](https://aave.com/docs/concepts/flash-loans)).
+2. Use the 5,002.5 USDT from the trade to fully repay the flash loan (including the fee).
+
+:::info
+
+Exchange ratios on the Sepolia network can be very inconsistent. To ensure successful execution of the order, a large amount of USDC is withdrawn from Aave and used in the trade.
+
+General considerations when testing flash loans on Sepolia and Aave:
+- [Token addresses in Aave may not correspond](https://github.com/mrnaveira/cow-flashloan-tester?tab=readme-ov-file#2-select-aave-tokens) with tokens names in other dapps.
+- [Token ratios](https://github.com/mrnaveira/cow-flashloan-tester?tab=readme-ov-file#4-determine-token-amounts-needed-to-trade-and-deposit) may need to be balanced, and trading amounts adjusted accordingly.
+- Tokens in Aave can be [obtained from the faucet](https://github.com/mrnaveira/cow-flashloan-tester?tab=readme-ov-file#5-get-tokens-from-the-faucet).
+- Check [withdrawal and borrowing availability](https://github.com/mrnaveira/cow-flashloan-tester?tab=readme-ov-file#7-check-aaves-withdrawalborrow-availability) of the tokens on Aave.
+
+In a production network (e.g. mainnet), these considerations generally won’t apply.
+
+:::
+
+## Aave repay pre-hook
+We need to define a [pre-hook](/cow-protocol/reference/core/intents/hooks) to repay the 5,000 USDT debt on Aave.
```

Comment:
Being a bit more explicit with our context would be good.
i.e. we need to repay the debt so that the Aave pool allows us to withdraw the collateral from it.